### ISSUE TITLE
Overview Issues http/transport

### ISSUE THREAD

**Issue by jralph** [lmenezes/cerebro#64]
Hi,

I'm having an issue getting Cerebro to connect to an es cluster running using Amazons Elasticserach Service. I know this service does not support the transport layer for es, does Cerebro use the the transport layer to display the overview and other features, or does it just use the http api?

Thanks!

**Comment by lmenezes** [comment]
hey @jralph 
it uses only the http api, so there should be no problem as long as cerebro is running on a host that has access to the instances. Do you have anything special on your amazon setup?

**Comment by jralph** [comment]
Nothing special, just a default setup.

The odd thing is, Cerebro is getting the cluster state (its green), and shows notifications when nodes are added/removed.

The navbar and cluster_changes http requests return ok, but the overview request returns with this message:

```
This exception has been logged with id 7304npjnc.
```

**Comment by lmenezes** [comment]
hum, do you have anything on the app logs?

**Comment by jralph** [comment]
This is the error reported in the application.log file:

```
@730518fko - Internal server error, for (POST) [/overview] ->

play.api.UnexpectedException: Unexpected exception[JsResultException: JsResultException(errors:List((,List(ValidationError(List('host' is undefined on object: {"name":"cZe2D_b","version":"5.1.1","build_hash":"5395e21","roles":["master","data","ingest"],"os":{"refresh_interval_in_millis":1000,"available_processors":1,"allocated_processors":1},"jvm":{"pid":4296,"start_time_in_millis":1485944559636,"mem":{"heap_init_in_bytes":1073741824,"heap_max_in_bytes":1065025536,"non_heap_init_in_bytes":2555904,"non_heap_max_in_bytes":0,"direct_max_in_bytes":1065025536},"using_compressed_ordinary_object_pointers":"true"}}),WrappedArray())))))]
	at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:261) ~[com.typesafe.play.play_2.11-2.4.6.jar:2.4.6]
	at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:191) ~[com.typesafe.play.play_2.11-2.4.6.jar:2.4.6]
	at play.api.GlobalSettings$class.onError(GlobalSettings.scala:179) [com.typesafe.play.play_2.11-2.4.6.jar:2.4.6]
	at play.api.DefaultGlobal$.onError(GlobalSettings.scala:212) [com.typesafe.play.play_2.11-2.4.6.jar:2.4.6]
	at play.api.http.GlobalSettingsHttpErrorHandler.onServerError(HttpErrorHandler.scala:94) [com.typesafe.play.play_2.11-2.4.6.jar:2.4.6]
Caused by: play.api.libs.json.JsResultException: JsResultException(errors:List((,List(ValidationError(List('host' is undefined on object: {"name":"cZe2D_b","version":"5.1.1","build_hash":"5395e21","roles":["master","data","ingest"],"os":{"refresh_interval_in_millis":1000,"available_processors":1,"allocated_processors":1},"jvm":{"pid":4296,"start_time_in_millis":1485944559636,"mem":{"heap_init_in_bytes":1073741824,"heap_max_in_bytes":1065025536,"non_heap_init_in_bytes":2555904,"non_heap_max_in_bytes":0,"direct_max_in_bytes":1065025536},"using_compressed_ordinary_object_pointers":"true"}}),WrappedArray())))))
	at play.api.libs.json.JsReadable$$anonfun$2.apply(JsReadable.scala:20) ~[com.typesafe.play.play-json_2.11-2.4.6.jar:2.4.6]
	at play.api.libs.json.JsReadable$$anonfun$2.apply(JsReadable.scala:20) ~[com.typesafe.play.play-json_2.11-2.4.6.jar:2.4.6]
	at play.api.libs.json.JsResult$class.fold(JsResult.scala:73) ~[com.typesafe.play.play-json_2.11-2.4.6.jar:2.4.6]
	at play.api.libs.json.JsError.fold(JsResult.scala:13) ~[com.typesafe.play.play-json_2.11-2.4.6.jar:2.4.6]
	at play.api.libs.json.JsReadable$class.as(JsReadable.scala:18) ~[com.typesafe.play.play-json_2.11-2.4.6.jar:2.4.6]
```

**Comment by jralph** [comment]
The es host format we're using in Cerebero is as follows: `http://my-es-domain.com:80` (Aws exposes port 80 and port 443 (Both the same as the standard es port 9200)

**Comment by lmenezes** [comment]
hum, this is odd. It seems that the response for /_nodes/_all/os,jvm is not returning all the information expected.

Could you paste the response of calling this api directly on one of your nodes(do of course remove any private information, I'm mostly interested in which fields are being returned, not so much on the content).

**Comment by jralph** [comment]
This is the data I'm getting from that endpoint.

```
{
   "_nodes": {
      "total": 2,
      "successful": 2,
      "failed": 0
   },
   "cluster_name": "!!removed!!",
   "nodes": {
      "cZe2D_bJQNK1dtplH7aMzQ": {
         "name": "cZe2D_b",
         "version": "5.1.1",
         "build_hash": "5395e21",
         "roles": [
            "master",
            "data",
            "ingest"
         ],
         "os": {
            "refresh_interval_in_millis": 1000,
            "available_processors": 1,
            "allocated_processors": 1
         },
         "jvm": {
            "pid": 4296,
            "start_time_in_millis": 1485944559636,
            "mem": {
               "heap_init_in_bytes": 1073741824,
               "heap_max_in_bytes": 1065025536,
               "non_heap_init_in_bytes": 2555904,
               "non_heap_max_in_bytes": 0,
               "direct_max_in_bytes": 1065025536
            },
            "using_compressed_ordinary_object_pointers": "true"
         }
      },
      "IVZ_FPfyQbaNq0yS7NTY4A": {
         "name": "IVZ_FPf",
         "version": "5.1.1",
         "build_hash": "5395e21",
         "roles": [
            "master",
            "data",
            "ingest"
         ],
         "os": {
            "refresh_interval_in_millis": 1000,
            "available_processors": 1,
            "allocated_processors": 1
         },
         "jvm": {
            "pid": 4233,
            "start_time_in_millis": 1485944573340,
            "mem": {
               "heap_init_in_bytes": 1073741824,
               "heap_max_in_bytes": 1065025536,
               "non_heap_init_in_bytes": 2555904,
               "non_heap_max_in_bytes": 0,
               "direct_max_in_bytes": 1065025536
            },
            "using_compressed_ordinary_object_pointers": "true"
         }
      }
   }
}
```

**Comment by jralph** [comment]
I think I see the issue myself. Compared the result above to a fresh install of es on its own vm (without aws) and the missing property is in the node block.....there is no `transport_address` or `host` property for each node.

**Comment by jralph** [comment]
I believe a simple fix to this problem would be in the `public/overview.html` file, changing line `199` to only display the host if the host property exists, maybe displaying "unknown" or something instead. This would then allow Cerebro to work with the AWS Elasticsearch service.

**Comment by lmenezes** [comment]
@jralph I will take a look at this later and try to create a fix. I will let you know :)

**Comment by jralph** [comment]
That would be brilliant, thanks. Let me know when you have something and need it testing with AWS and I'll give things a run through.

**Comment by jralph** [comment]
Hi,

Is there any update on this at all?

**Comment by jralph** [comment]
Hi @lmenezes,

Thanks for the fix. It appears the same thing is happening with the jvm_version also.

Fixed it by adding the following to `app/models/overview/Node.scala`

```
val jvm_version = (info \ "jvm" \ "version").asOpt[JsString].getOrElse(JsNull)

// Within the Json.obj...
"jvm_version" -> jvm_version,
```

**Comment by lmenezes** [comment]
@jralph great. i have no instance running on aws, so no real way to test this. Will also add this exception there then. 👍

**Comment by jralph** [comment]
No problem. I've never written anything in scala before so I didn't know how to test it myself. Since looking at your last commit I was able to test this one...and I can verify that everything is working with aws once the jvm_version is fixed.

**Comment by nickmaccarthy** [comment]
I am also running into this issue as well when trying to view an Amazon ES distribution.  Cerebro would be great to use for the AWS Elasticsearch service since their front end is poor to say the least.  Has this fix been added to a recent version?

**Comment by jralph** [comment]
@nickmaccarthy Personally I've been running this by pulling down the master branch and running the command `sbt run`.  It's pretty simple, install Java, install SBT, run `sbt run` within the repository.

Would still be good to have this released officially though.

**Comment by lmenezes** [comment]
hey,

this is available since v0.6.0(although latest is 0.6.1). If this is still not working, please reopen this with the error :)
